### PR TITLE
fix: Remove char set from content headers for JSON payloads

### LINKED ISSUE #910: Contacts API fails with Content-Type application/json; charset=utf-8
Making a request to the Contacts API endpoint (`{baseUrl}/marketing/contacts`) with a Content-Type of `application/json; charset=utf-8` returns a 400 Bad Request saying that the content type is not supported:

```
{
    "errors": [
        {
            "field": "content-type",
            "message": "invalid content-type: application/json; charset=utf-8"
        }
    ]
}
```

You can try it out yourself at https://sendgrid.com/docs/API_Reference/api_v3.html and send a PUT request to the Contacts endpoint.

Unfortunately this C# client ALWAYS sets the charset to UTF8, which makes it unusable for some endpoints like the one above. Can the SendGrid folks please accept the entirely valid media type with the charset or at least make it optional in the RequestAsync method of the client?

Feels really broken right now and is frustrating that I've spend a couple hours trying to understand why a valid request gets denied for appending the charset -.-

If I send the same request with just `application/json` (without the utf-8 charset) then the request succeeds, but that has to be done manually and can't be done with the RequestAsync method at the moment.

### PR DESCRIPTION
**Pull request by marius-stanescu**
Fixes #910 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the development branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Removes the character set from the content headers

### CODE REVIEW

File: src/SendGrid/SendGridClient.cs [hunk 1]
```diff
@@ -228,6 +228,7 @@ public virtual async Task<Response> MakeRequest(HttpRequestMessage request, Canc
                 RequestUri = new Uri(baseAddress, this.BuildUrl(urlPath, queryParams)),
                 Content = requestBody == null ? null : new StringContent(requestBody, Encoding.UTF8, this.MediaType)
             };
+            request.Content.Headers.ContentType.CharSet = null;
```
Comment (thinkingserious) [inline src/SendGrid/SendGridClient.cs]: It looks like this is causing many breaks in the integration tests.

My suggestion would be to implement it as an option with the default being as-is. Ideally, we'd set this only for the relevant endpoints, but I think as a first start, making this configurable is a good step. 

Alternatively, we can simply document how to pass in your own custom client that specifically updates this setting.

Comment (hallidev) [reply src/SendGrid/SendGridClient.cs]: I believe this needs to be fixed server-side. Sending the CharSet from the client here is perfectly valid. This is a workaround at best. The problem is that the server is rejecting valid requests.

**Comment by childish-sambino** [comment]
Closing this as the consensus is that this is a server-side issue that the client shouldn't need a workaround/hack to resolve. Recommend contacting [support](http://support.sendgrid.com) to investigate a back-end fix.

File: src/SendGrid/SendGridClient.cs (hunk 1 above)
Comment (childish-sambino) [reply src/SendGrid/SendGridClient.cs]: See #910 for follow-up regarding the server-side issue.

**Pull request merged**